### PR TITLE
chore: Fluent API for `queryOne` and `queryFirst`

### PR DESCRIPTION
**Pull request by sharat87**
## Summary by CodeRabbit

- **Refactor**
	- Improved query building and execution methods across various repositories for enhanced clarity and efficiency.
	- Introduced a more explicit handling of optional parameters in query methods.
- **Bug Fixes**
	- Adjusted method calls for building queries and handling permissions to ensure more accurate results and access control.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public interface AppsmithRepository<T extends BaseDomain> {
 
     Mono<T> updateById(String id, T resource, AclPermission permission);
 
-    QueryAllParams<T> queryAll();
+    QueryAllParams<T> buildQuery();
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java:28]: Just a tiny bit of feedback on nomenclature, can we call this `queryBuilder`? Helps the builder pattern muscles kick in.

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java:28]: The current naming I'd associate more with the terminal step.

**Review by nidhi-nair — COMMENTED** [review]
LGTM, nomenclature suggestion is all.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java (hunk 1 above)
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java]: Fair point. Done.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**